### ISSUE TITLE
non-wasm platform specific files are seen by wasm backend

### ISSUE THREAD

**Issue by Avey777** [vlang/v#19228]
### Describe the bug

 v -b wasm main.v 
 
Compilation error

### Reproduction Steps

```
module main

import veb

struct App {
	veb.Middleware[Context]
}

struct Context {
	veb.Context
}

fn main() {
	mut app := &App{}
	veb.run[App, Context](mut app, 8082)
}

@['/api'; get; post]
pub fn (mut app App) index(mut ctx Context) veb.Result {
	return ctx.json([
		'code: 200,
        message: success',
	])
}
```

### Expected Behavior

Generate main.wasm executable file

### Current Behavior

```
/home/Jengro/.vmr/versions/v_versions/v_latest/vlib/net/address_darwin.c.v:45:12: error: cannot register struct `net.Ip6`, another type with this name exists
   43 |
   44 | @[_pack: '1']
   45 | pub struct Ip6 {
      |            ~~~
   46 |     port      u16
   47 |     flow_info u32
```

### V version

V 0.4.0 34842b1

### Environment details (OS name and version, etc.)

os： windows10

**Comment by JalonSolov** [comment]
Something odd... it shouldn't be trying to compile a windows specific file in a wasm compile.

**Comment by JalonSolov** [comment]
I agree - platform specific files should be handled long before the backend is hit.

**Comment by Avey777** [comment]
> The backend is not respecting the arch specific files, this has been known for a while. In reality, this should be handled by the builder without being backend specific. As the backend isn't able to work with most library imports, and I have chosen not to implement much changes before the new compiler infrastructure, this may go unfixed for some time.`wasm``wasm`

Wasm is oriented towards the future, which is very important.

Is it possible for me to see this issue fixed within the next 3 months?

**Comment by JalonSolov** [comment]
While anything it possible, I wouldn't count on it.  This issue has to wait until the new compiler infrastructure changes are completed, and the time for that is unknown.

Perhaps we'll all be pleasantly surprised, but don't be upset otherwise.

**Comment by Avey777** [comment]
Is the new compiler work underway? Where can I follow the progress of compiler development?

**Comment by JalonSolov** [comment]
https://www.twitch.tv/v_language

**Comment by Avey777** [comment]
https://github.com/vlang/v/commit/7519f915ffdafd3229fdd8912f7d78481614cb35

Does this mean that the V compiler has begun to make progress?

**Comment by medvednikov** [comment]
It's been making progress for a while, it's just been merged into master.

**Comment by Avey777** [comment]
> It's been making progress for a while, it's just been merged into master.

I will continue to pay attention and look forward to fixing this issue

**Comment by Avey777** [comment]
> The backend is not respecting the arch specific files, this has been known for a while. In reality, this should be handled by the builder without being backend specific. As the backend isn't able to work with most library imports, and I have chosen not to implement much changes before the new compiler infrastructure, this may go unfixed for some time.`wasm``wasm`

Now that tinyv has been merged into the main branch, can this problem be solved?

**Comment by Avey777** [comment]
Is there any progress starting?

**Comment by Avey777** [comment]
V0.4.4
```
module main

import vweb

const (
	port = 8082
)

struct App {
	vweb.Context
}

pub fn (app App) before_request() {
	println('[web] before_request: ${app.req.method} ${app.req.url}')
}

fn main() {
	mut app := &App{}
	vweb.run(app, port)
}

@[get]@[post]@["/api"]
pub fn (mut app App) index() vweb.Result {

	return app.json([
        "code: 200,
        message: success"

    ])
}

```
new err
```
PS E:\Documents\Gitea\vprod\src> v -b wasm .\main.v
C:/Unpack/v/vlib/os/file.js.v:3:12: error: cannot register struct `os.File`, another type with this name exists
    1 | module os
    2 |
    3 | pub struct File {
      |            ~~~~
    4 | pub:
    5 |     fd int
PS E:\Documents\Gitea\vprod\src>
```

**Comment by JalonSolov** [comment]
No, there haven't been any changes to the wasm backend in a while.

**Comment by Avey777** [comment]
> The backend is not respecting the arch specific files, this has been known for a while. In reality, this should be handled by the builder without being backend specific. As the backend isn't able to work with most library imports, and I have chosen not to implement much changes before the new compiler infrastructure, this may go unfixed for some time.`wasm``wasm`

The new compiler has been merged into the main branch. Can we try to solve this problem in the new year?

**Comment by JalonSolov** [comment]
Yes, it's in the main branch.  However, it is still separate, in separate subdirs (`cmd/v2` and `vlib/v2`).  None of the new code is compiled or used by default, yet.

**Comment by Avey777** [comment]
```
import os
import db.pg



const pg_host = os.getenv_opt('PGHOST') or { '***' }
const pg_user = os.getenv_opt('PGUSER') or { '***' }
const pg_pass = os.getenv_opt('PGPASS') or { '***' }
const pg_db = os.getenv_opt('PGDATABASE') or { '***' }



fn psql() ! {
	eprintln('------------ start -----------------')
	mut db := pg.connect(host:pg_host, user: pg_user, password: pg_pass, dbname: pg_db)!

	db.exec_one('TRUNCATE TABLE events RESTART IDENTITY;') or { eprintln(err) }

}

fn main() {
	psql()!
	eprintln('------------ end -------------------')
}
```

```
(base) root@Jengro:/opt/redash# v -b wasm -autofree -gc  none main.v 
/opt/v/vlib/os/file.js.v:3:12: error: cannot register struct `os.File`, another type with this name exists
    1 | module os
    2 | 
    3 | pub struct File {
      |            ~~~~
    4 | pub:
    5 |     fd int
(base) root@Jengro:/opt/redash# 
```

@JalonSolov
Is this question the same as the previous one?

**Comment by JalonSolov** [comment]
Yes.  There have not been any changes regarding this issue.

**Comment by Avey777** [comment]
> The `wasm` backend is not respecting the arch specific files, this has been known for a while. In reality, this should be handled by the builder without being backend specific. As the `wasm` backend isn't able to work with most library imports, and I have chosen not to implement much changes before the new compiler infrastructure, this may go unfixed for some time.

Is this planned again now? Wait until V2 is finished, right?

**Comment by JalonSolov** [comment]
Yes, wait until V2 is finished.

**Comment by Avey777** [comment]
> Yes, wait until V2 is finished.


This will be a very long wait.❤️

**Comment by Avey777** [comment]
我的 issue ，有没有做无作为，但是请@spytheman   ，这个说话方式不讨人喜欢的家伙， 发表任何评论

**Comment by jorgeluismireles** [comment]
> has the highest number of individual contributions to V

I wrote a little program to inspect V git results with sqlite:
```
* Filter subject like (% fix%)
* All filtered commits:
	6341
* User all commits (like "Alex"):
	3767
* By user (min 50):
	['1283', 'Delyan Angelov']
	['1200', 'yuyi']
	['909', 'Alexander Medvednikov']
	['688', 'Felipe Pena']
	['150', 'shove']
	['138', 'joe-conigliaro']
	['116', 'Turiiya']
	['96', 'Daniel Däschle']
	['70', 'Joe Conigliaro']
	['65', 'Larpon']
	['59', 'penguindark']
	['51', 'Uwe Krüger']
* By date filtered (min 15):
	['29', '2021-04-05']
	['22', '2022-04-15']
	['21', '2020-04-02']
	['20', '2020-04-08']
	['16', '2020-12-28']
* Last filtered (5):
	2025-05-01 Delyan Angelov	tools: fix `./v -d network test cmd/tools/vpm`
	2025-05-01 Delyan Angelov	tools: fix `v timeout 2 sleep 5`
	2025-04-30 Eliyaan (Nopana)	tools: vreduce fix var names (#24373)
	2025-04-30 Delyan Angelov	ci: fix hub_docker_ci.yml issue - comment out the cache-from:/cache-to: lines again
	2025-04-30 kbkpbot	os: fix windows rmdir GetLastError() (fix #24356) (#24357)
```

**Comment by Avey777** [comment]
> "this unappealing guy" has the highest number of individual contributions to V.  Click the "Contributor" link on the main repo page to see that he has made more First of all, thank you very much for your enthusiasm, whether this or other issues.
However, I suggest that you do not directly say that this is not the way to work with open source, or you can express it in a slightly indirect way, as being too direct will make people feel that you are coming to teach them how to do things, which will make them uncomfortable.
If you don't want to solve the bug, you can ignore it, or you can say how much you need to pay to do it. But if you don't have anything to do with it, please don't say things like this is not an open source way,

**Comment by Jengro777** [comment]
Due to the abandonment of vweb, we are now using veb for feedback

```
module main

import veb

struct App {
	veb.Middleware[Context]
}

struct Context {
	veb.Context
}

fn main() {
	mut app := &App{}
	veb.run[App, Context](mut app, 8082)
}

@['/api'; get; post]
pub fn (mut app App) index(mut ctx Context) veb.Result {
	return ctx.json([
		'code: 200,
        message: success',
	])
}
```
